### PR TITLE
fixes SQLite3 compatibility issue for test env. Closes #966

### LINKED ISSUE #966: failing tests due to incorrect SQLite version
When I run the test suite locally, I am getting 7 failures caused by the sqlite3 adapter version mismatch. 
```shell 
  1) suspenders:json generates the Gemfile for JSON parsing
     Failure/Error: expect("Gemfile").to match_contents(%r{gem .oj.})
       expected "Gemfile" to match contents /gem .oj./
     # ./spec/features/json_spec.rb:7:in `block (2 levels) in <top (required)>'

  2) suspenders:production:deployment generates the configuration for a production deployment
     Failure/Error: expect("bin/deploy").to exist_as_a_file
       expected "bin/deploy" to exist as a file
     # ./spec/features/production/deployment_spec.rb:9:in `block (2 levels) in <top (required)>'

  3) suspenders:production:email generates the configuration for a production email deployment
     Failure/Error: contents = IO.read(File.join(project_path, filename))

     Errno::ENOENT:
       No such file or directory @ rb_sysopen - /Users/---/Desktop/ruby_sandbox/suspenders/tmp/dummy_app/config/smtp.rb
     # ./spec/support/match_contents_matcher.rb:3:in `read'
     # ./spec/support/match_contents_matcher.rb:3:in `block (2 levels) in <top (required)>'
     # ./spec/features/production/email_spec.rb:7:in `block (2 levels) in <top (required)>' 

  4) suspenders:production:email destroys the configuration for a production email deployment
     Failure/Error: json = JSON.parse(IO.read(filepath), symbolize_names: true)

     Errno::ENOENT:
       No such file or directory @ rb_sysopen - /Users/---Desktop/ruby_sandbox/suspenders/tmp/dummy_app/app.json
     # ./spec/support/contain_json_matcher.rb:11:in `read'
     # ./spec/support/contain_json_matcher.rb:11:in `block (2 levels) in <top (required)>'
     # ./spec/features/production/email_spec.rb:38:in `block (2 levels) in <top (required)>'

  5) suspenders:production:manifest generates the manifest for a production build
     Failure/Error: json = JSON.parse(IO.read(filepath), symbolize_names: true)

     Errno::ENOENT:
       No such file or directory @ rb_sysopen - /Users/---/Desktop/ruby_sandbox/suspenders/tmp/dummy_app/app.json
     # ./spec/support/contain_json_matcher.rb:11:in `read'
     # ./spec/support/contain_json_matcher.rb:11:in `block (2 levels) in <top (required)>'
     # ./spec/features/production/manifest_spec.rb:7:in `block (2 levels) in <top (required)>'

  6) suspenders:production:manifest destroys the manifest for a production build
     Failure/Error: json = JSON.parse(IO.read(filepath), symbolize_names: true)

     Errno::ENOENT:
       No such file or directory @ rb_sysopen - /Users/---/Desktop/ruby_sandbox/suspenders/tmp/dummy_app/app.json
     # ./spec/support/contain_json_matcher.rb:11:in `read'
     # ./spec/support/contain_json_matcher.rb:11:in `block (2 levels) in <top (required)>'
     # ./spec/features/production/manifest_spec.rb:23:in `block (2 levels) in <top (required)>'

  7) suspenders:staging:pull_requests generates the configuration for Heroku pipeline review apps
     Failure/Error:
       expect("config/environments/production.rb").to \
         match_contents(%r{HEROKU_APP_NAME})

       expected "config/environments/production.rb" to match contents /HEROKU_APP_NAME/
     # ./spec/features/staging/pull_requests_spec.rb:7:in `block (2 levels) in <top (required)>'
```

and the trace is:
```shell
.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bundler-2.0.1/lib/bundler/rubygems_integration.rb:408:in `block (2 levels) in replace_gem': Error loading the 'sqlite3' Active Record adapter. Missing a gem it depends on? can't activate sqlite3 (~> 1.3.6), already activated sqlite3-1.4.0. Make sure all dependencies are added to Gemfile. (LoadError)
	from /Users/---/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/activerecord-5.2.2/lib/active_record/connection_adapters/sqlite3_adapter.rb:12:in `<main>'
	from /Users/---/.rbenv/versions/2.5.3/lib/ruby/gems/2.5.0/gems/bootsnap-1.3.2/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:21:in `require'
```

As pointed out in this PR: https://github.com/thoughtbot/factory_bot_rails/pull/318, the current `rails new` generator defaults to the most recent version of SQLite, which is incompatible with Rails.

### PR DESCRIPTION
**Pull request by dimanyc**
freezes `SQLite` version to 1.36 in `run_suspenders` helper

paired with @composerinteralia

### CODE REVIEW

**Comment by composerinteralia** [comment]
I think it would be helpful to link to the issue in the commit message. I can imagine my future self wondering why we locked to a specific version.

**Comment by dimanyc** [comment]
> I think it would be helpful to link to the issue in the commit message. I can imagine my future self wondering why we locked to a specific version.

good point. Commit message updated

**Comment by mike-burns** [comment]
The upstream issue this solves: https://github.com/rails/rails/issues/35153

With the fix: https://github.com/rails/rails/pull/35154

Why is our fix different from Rails'?

**Comment by dimanyc** [comment]
I've the master branch with [5.2.2](https://rubygems.org/gems/rails/versions/5.2.2) and the issue still occurs. I think we should go with this fix for now, and revisit if if / when https://github.com/rails/rails/pull/35154 is released.

**Comment by composerinteralia** [comment]
@mike-burns it doesn't look that change was released yet. If we merge this, we should be able to revert the commit once 5.2.3 gets released.

**Comment by mike-burns** [comment]
Oh I guess this is the fix: https://github.com/rails/rails/pull/35169

But again: why is the Rails fix this:

```ruby
gem "sqlite3", "'~> 1.3', '>= 1.3.6'"
```

and our fix this:

```ruby
gem "sqlite3", "~> 1.3.6"
```

? What am I missing?

**Comment by composerinteralia** [comment]
Ah, I think I read your question as "why do we need to fix this in suspenders when it is already fixed in Rails?" instead of "why is our fix different than the fix in Rails?"

I think we were looking at https://github.com/thoughtbot/factory_bot_rails/pull/318. I merged that PR because it worked. We should probably use `"'~> 1.3', '>= 1.3.6'"` instead.

**Comment by dimanyc** [comment]
@mike-burns it seems like `'>= 1.3.6'` will raise an exception about adapter incompatibility:

```bash wrap
Error loading the 'sqlite3' Active Record adapter. 
Missing a gem it depends on? can't activate sqlite3 (~> 1.3.6), 
already activated sqlite3-1.4.0. 
Make sure all dependencies are added to Gemfile. (LoadError)
```
switching from:
```ruby
gem "sqlite3", "~> 1.3", "> 1.3.6"
```
to:
```ruby
gem "sqlite3", "~> 1.3", "< 1.4"
```
fixes the issue. It is something with my local set up?

**Comment by mike-burns** [comment]
Hm! Not sure. I don't fully understand why Rails went with that fix; if we have one that works more reliably, I suppose we should use it.

**Comment by composerinteralia** [comment]
I think I understand this now. 

In current versions of Rails:

* The default Gemfile doesn't have any version constraints for sqlite3, so we end up with the newest version 1.4
* The Rails sqlite3 adapter specifies `gem "sqlite3", "~> 1.3.6"` https://github.com/sponomarev/rails/blob/5-2-2/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L12
* To fix, we need to match the version constraint in the sqlite3 adapter. This matches the fix used in Rails for the bug report templates https://github.com/rails/rails/pull/35157/files

---

In future versions of Rails:

* The sqlite3 adapter version constraints were changed in https://github.com/rails/rails/pull/35154/files to allow version 1.4 -> `gem "sqlite3", "'~> 1.3', '>= 1.3.6'"`
* The default Gemfile was changed to match those consraints in https://github.com/rails/rails/pull/35169/files

---

In short, I think our version constraint here is correct for the current versions of Rails.

**Comment by composerinteralia** [comment]
These tests started passing for me once I merged in https://github.com/thoughtbot/suspenders/pull/972. I don't understand why, but maybe we don't need this anymore.

**Comment by composerinteralia** [comment]
5.2.3 is out, so we should be able to close this PR now.

**Comment by composerinteralia** [comment]
Confirmed we were able to get a passing build on master without this change. Closing
